exist-versioning-release.md

```diff
@@ -208,101 +313,169 @@ You will require a system with:
     What is the new development version for "eXist-db"? (org.exist-db:exist) 5.4.0-SNAPSHOT: :
     ```
 
-6.  Once the prepare process completes you can perform the release. This will upload Maven Artifacts to Maven
-Central (staging), Docker images to Docker Hub, and eXist-db distributions and installer to Github releases:
-    ```
-    $ mvn -Ddocker=true -Dmac-signing=true -Djarsigner.skip=false -Darguments="-Ddocker=true -Dmac-signing=true -Djarsigner.skip=false" release:perform
-    ```
+5. Test the release artefacts as much as possible on different Operating systems (Mac, Windows, Linux).  
+
+   * `$EXIST/exist-distribution/target/` 
+     * eXist-db-<version>.dmg
+     * exist-distribution-eXist-db-<version>.bz2
+     * exist-distribution-eXist-db-<version>-win.zip
+   * `$EXIST/exist-installer-<version>.jar` 
+
+#### Signing Binaries (work in progress)
+
+In order to allow end-users to run the installer as well as the application on macOSX (Catalina) the application needs to be notarized by Apple.
+    
+At this step the application is ready to be notarized and the notarisation ticket must be stapled to the app once that was successful.
+    
+#### Troubleshooting
+
+##### Cleanup dry-run
+
+    mvn \
+      -Pdocker \
+      -Pinstaller \
+      -Pconcurrency-stress-tests \
+      -Pmicro-benchmarks \
+      release:clean
+
+##### Rollback Release
+
+If you need to rollback the `mvn release:prepare` phase:
 
-7.  Update the stable branch (`master`) of eXist-db to reflect the latest release:
+`mvn release:rollback` works but
+
+- the two commits will remain in the git history and
+- you will need to remove git tag from github by hand    
+    
+    
+### Releasing the Product 
+    
+1. Once the prepare process completes successfully and artifacts have been tested you can perform the release. This will upload Maven Artifacts to Maven Central (staging), Docker images to Docker Hub, and eXist-db distributions and installer to Github releases. 
+    
+    ```bash
+    mvn \
+      -Pdocker \
+      -Pinstaller \
+      -Pconcurrency-stress-tests \
+      -Pmicro-benchmarks \
+      -Darguments="-Djarsigner.skip=false -Dmac-signing=true -Dmac.codesign.identity=\"your codesign ID\"" \
+      release:perform
     ```
-    $ git push origin eXist-5.3.0:master
+
+    #### Notarizing macOSX installer (work in progress)
+    
+    Now the installer in `target/checkout/exist-distribution/eXist-5.3.0.dmg` needs to be notarized and the notarisation ticket must be stapled to the installer once that was successful.
+    The notarized installer containing the notarized app must then be uploaded to the github releases page.
+
+    
+2. Update the stable branch (`master`) of eXist-db to reflect the latest release:
```

Comment:
Before this... A visual checklist of things needs to be completed. Was the output of the `release:perform` step as expected. Do the new version numbers look correct locally? do they look correct on the exist-db GitHub? Does the tag look correct, did the tag get pushed to the exist-db GitHub? Has at least one other person also checked such things and also confirmed this to you?

Also before modifying `master`, I would suggest that the oss.sonatype.org artifacts have to be released successfully. Sonatype has its own suit of checks, and if the artifacts fail those for some reason, then you have to undo the release process, fix the issue, go all the way back to the start of the release process and start again. So it is important that Sonatatype artifacts are published OK before we change master. Changing master should really be the last thing that ever happens (apart from perhaps tweeting/emailing/etc)

Reply:
Interesting. This sounds like very important information that belongs in this guide.